### PR TITLE
docs: fix new clippy lint about excessively long initial line

### PR DESCRIPTION
**Pull request by apoelstra**
Also does a few miscellaneous fixes that I noticed while fixing the lint.

### CODE REVIEW

**Review by tcharding — COMMENTED** [review]
Ha, Poelstra patching docs like a junior dev! :)

I rekon we should fix the column width of the whole first paragraph not just move the line.

File: src/miniscript/types/malleability.rs [hunk 1]
```diff
@@ -12,11 +14,15 @@ pub enum Dissat {
...
     /// of `Dissat::Unique` and `Input::Zero` implies that a fragment is
     /// impossible to satisfy (is a `0` or equivalent).
     Unique,
-    /// No assumptions may be made about dissatisfying this fragment. This
+    /// No assumptions may be made about dissatisfying this fragment.
+    ///
+    /// This
     /// does not necessarily mean that there are multiple dissatisfactions;
     /// there may be none, or none that are always available (e.g. for a
     /// `pk_h` the key preimage may not be available).
```
Comment (tcharding) [inline src/miniscript/types/malleability.rs:28]: This is an example that would be better with the column width fixing.

**Comment by apoelstra** [comment]
We could go through all the docs and make them look nice, yes.

But this is causing local CI failures (and, I think, real CI failures) so I would like to just get it working.

**Comment by tcharding** [comment]
Sure thing.

**Comment by tcharding** [comment]
ACK since this is not `rust-bitcoin` anyway, so I'm not as invested in it and the docs are not as polished.

File: src/psbt/finalizer.rs [hunk 2]
```diff
@@ -351,7 +352,9 @@ fn interpreter_inp_check<C: secp256k1::Verification, T: Borrow<TxOut>>(
     Ok(())
 }
 
-/// Finalize the psbt. This function takes in a mutable reference to psbt
+/// Finalize the psbt.
```
Comment (storopoli) [inline src/psbt/finalizer.rs:355]: Maybe
```suggestion
/// Finalizes the psbt.
```

to adhere with other docstrings?

File: src/plan.rs [hunk 3]
```diff
@@ -67,7 +69,9 @@ pub trait AssetProvider<Pk: MiniscriptKey> {
     }
 
     /// Given a keyhash, look up the EC signature and the associated key.
```
Comment (storopoli) [inline src/plan.rs:71]:
```suggestion
    /// Given a keyhash, looks up the EC signature and the associated key.
```

File: src/plan.rs [hunk 4]
```diff
@@ -1,7 +1,9 @@
 // SPDX-License-Identifier: CC0-1.0
 
-//! A spending plan or *plan* for short is a representation of a particular spending path on a
-//! descriptor. This allows us to analayze a choice of spending path without producing any
+//! A spending plan (or *plan*) is a representation of a particular spending path on a
+//! descriptor.
+//!
+//! This allows us to analayze a choice of spending path without producing any
```
Comment (storopoli) [inline src/plan.rs:6]:
```suggestion
//! This allows us to analyze a choice of spending path without producing any
```

File: src/plan.rs [hunk 5]
```diff
@@ -76,7 +80,9 @@ pub trait AssetProvider<Pk: MiniscriptKey> {
     }
 
     /// Given a keyhash, look up the schnorr signature and the associated key.
```
Comment (storopoli) [inline src/plan.rs:82]:
```suggestion
    /// Given a keyhash, looks up the Schnorr signature and the associated key.
```

File: src/plan.rs [hunk 6]
```diff
@@ -67,7 +69,9 @@ pub trait AssetProvider<Pk: MiniscriptKey> {
     }
 
     /// Given a keyhash, look up the EC signature and the associated key.
+    ///
     /// Returns the key if a signature is found.
+    ///
     /// Even if signatures for public key Hashes are not available, the users
```
Comment (storopoli) [inline src/plan.rs:75]: why capital H?

File: src/miniscript/satisfy.rs [hunk 7]
```diff
@@ -58,7 +59,8 @@ pub trait Satisfier<Pk: MiniscriptKey + ToPublicKey> {
     /// Given a raw `Pkh`, lookup corresponding [`bitcoin::secp256k1::XOnlyPublicKey`]
```
Comment (storopoli) [inline src/miniscript/satisfy.rs:59]: I cannot annotate above but in L40 there's a "look" -> "looks" and "schnorr" -> "Schnorr"

**Review by storopoli — COMMENTED** [review]
Some comments.

**Comment by tcharding** [comment]
This is a CI fix PR @storopoli, also the `minisicript` rustdocs are no where near as polished as the `rust-bitcoin` docs so your suggestions, while valid, are probably to refined for this repo.

**Comment by apoelstra** [comment]
Indeed. There are lots of poorly-formatted and ungrammatical docs in this repo. Many of them apply to error enums and other types which themselves are a huge mess. I'd rather not change stuff unless we're fixing the underlying type, because it would be a lot of (potentially-wasted) work and extra rebasing for me.

**Comment by tcharding** [comment]
I never did get around to putting 3 months into `miniscript` - hopefully one day I will.

File: src/plan.rs (hunk 4 above)
Comment (storopoli) [reply src/plan.rs:6]: Ok can I review-beg this typo at least 😢

**Review by sanket1729 — APPROVED** [review]
ACK 05dace09f4f99c673fdb6b6fb35e6e0fcb134a68

**Comment by apoelstra** [comment]
At some point I will open an issue to really fix the docs; but as mentioned, right now the majority of the documentation by volume seems to be on error variants, and I hope to significantly change those over the coming months and years.

**Pull request merged**